test/DynamoCoreTests/DSCoreDataTests.cs

```diff
@@ -0,0 +1,199 @@
+            AssertPreviewValue(testNode.GUID.ToString(), expectedOutputs);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void StringifyJSON()
+        {
+            // Open the test graph
+            string path = Path.Combine(TestDirectory, @"core\json\JSON_Nodes_Test.dyn");
+            OpenModel(path);
+
+            // Get node data
+            WorkspaceModel workspace = CurrentDynamoModel.CurrentWorkspace;
+            string testNodeGuid = "31f973f9-0764-47e1-b770-7b6b97d8803e";
+            NodeModel testNode = getNodeById(workspace, testNodeGuid);
+
+            // Expected parsed types
+            List<string> expectedOutputs = Enumerable.Repeat(typeof(System.String).FullName, 8).ToList();
+            
+            // Verify node output types match expected output
+            AssertPreviewValue(testNode.GUID.ToString(), expectedOutputs);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void VerifyStringifyReplication()
+        {
+            // Load test graph
+            string path = Path.Combine(TestDirectory, @"core\json\JSON_Nodes_Replication.dyn");
+            OpenModel(path);
+
+            // Verify applying replication produces different result
+            AssertPreviewValue("b69f2b6e-a328-43cf-a7a3-db42576ce814", true);
+
+            // Verify DesignScript and node stringify functionality produces the same results - no replication.
+            AssertPreviewValue("e81d053f-dbc5-4b1e-86df-075c9b279aa3", true);
+
+            // Verify DesignScript and node stringify functionality produces the same results - with replication.
+            AssertPreviewValue("a0af3136-aaaf-40c2-b2ef-cee522f9ea45", true);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void VerifyDesignScriptMatchesNodes()
+        {
+            // Load test graph
+            string path = Path.Combine(TestDirectory, @"core\json\JSON_Nodes_DesignScript.dyn");
+            OpenModel(path);
+
+            // Verify DesignScript usage of ParseJSON and StringifyJSON results match nodes
+            AssertPreviewValue("5df5ed6d-6270-4018-a479-4f7cefcf7fe8", true);
+        }
+
+        [Test]
+        [Category("UnitTests")]
+        public void ParsingJSONInPythonReturnsSameResult()
+        {
+            // Load test graph
+            string path = Path.Combine(TestDirectory, @"core\json\JSON_Nodes_PythonJSONParsing.dyn");
+            OpenModel(path);
+
+            // Verify keys match when parsing JSON via Python
+            AssertPreviewValue("e4e600d9-12a6-400e-adb3-02c1ad26cddf", true);
+
+            // Verify values match when parsing JSON via Python
+            AssertPreviewValue("cdad5bf1-f5f7-47f4-a119-ad42e5084cfa", true);
+        }
+
+        // Helper Functions //
+
+        // Get node from a provided workspace by searching GUIDs
+        private NodeModel getNodeById(WorkspaceModel workspace, string guid)
```

Comment:
I think this already exists in the test class

Reply:
thanks, found it `workspace.NodeFromWorkspace(GUID)`